dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -2160,16 +2204,20 @@ public void build() {
 
       buildSelector();
 
-      // Build CFG for the cases.
-      extendWithNode(
-          new MarkerNode(
-              switchTree,
-              "start of switch statement #" + TreeUtils.treeUids.get(switchTree),
-              env.getTypeUtils()));
+      buildSwitchExpressionVar();
+
+      if (switchTree.getKind() == Kind.SWITCH) {
```

Comment:
After this line, I would add a comment "// It's a switch statement, not a switch expression.".
Likewise elsewhere.